sycl/unittests/config/ConfigTests.cpp

```diff
@@ -0,0 +1,231 @@
+                             "or does not contain them at all")));
+  } catch (...) {
+    FAIL() << "Check variable value bigger than MAX_CONFIG_VALUE failed";
+  }
+
+  // Check variable without value
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "a=" << std::endl;
+    File.close();
+  }
+  try {
+    sycl::detail::SYCLConfig<sycl::detail::SYCL_DEVICE_ALLOWLIST>::get();
+    throw std::logic_error("sycl::exception didn't throw");
+  } catch (sycl::exception &e) {
+    EXPECT_TRUE(std::regex_match(
+        e.what(), std::regex("The value contains more than ([\\d]+) characters "
+                             "or does not contain them at all")));
+  } catch (...) {
+    FAIL() << "Check variable without value failed";
+  }
+
+  // Check incorrect complex config processing
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "#\n   #\r\n#a=b\n\n\na\r\n aaa \r\na=b\r\na=\r" << std::endl;
+    File.close();
+  }
+  try {
+    sycl::detail::SYCLConfig<sycl::detail::SYCL_DEVICE_ALLOWLIST>::get();
+    throw std::logic_error("sycl::exception didn't throw");
+  } catch (sycl::exception &e) {
+    EXPECT_TRUE(std::regex_match(
+        e.what(), std::regex("The value contains more than ([\\d]+) characters "
+                             "or does not contain them at all")));
+  } catch (...) {
+    FAIL() << "Check incorrect complex config processing failed";
+  }
+
+  // Check simple correct config processing
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "SYCL_PRINT_EXECUTION_GRAPH=before_addCG" << std::endl;
+    File.close();
+  }
+  sycl::detail::readConfig(true);
+  EXPECT_EQ(std::string("before_addCG"),
+            sycl::detail::SYCLConfig<
+                sycl::detail::SYCL_PRINT_EXECUTION_GRAPH>::get());
+
+  // Check correct config processing
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "#a\r\n # b \r\n #a=b\r\n # a = "
+            "b\r\n\r\nSYCL_PRINT_EXECUTION_GRAPH=after_addCG\r\n"
+         << std::endl;
+    File.close();
+  }
+  sycl::detail::readConfig(true);
+  EXPECT_EQ(std::string("after_addCG"),
+            sycl::detail::SYCLConfig<
+                sycl::detail::SYCL_PRINT_EXECUTION_GRAPH>::get());
+
+  // Check multi assignment
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "a=b\r\nb=c\nSYCL_PRINT_EXECUTION_GRAPH=before_addCopyBack\r\nc=d";
+    File.close();
+  }
+  sycl::detail::readConfig(true);
+  EXPECT_EQ(std::string("before_addCopyBack"),
+            sycl::detail::SYCLConfig<
+                sycl::detail::SYCL_PRINT_EXECUTION_GRAPH>::get());
+
+  // Check assignment with comment
+  File.open("conf.txt");
+  if (File.is_open()) {
+    File << "SYCL_PRINT_EXECUTION_GRAPH=after_addCopyBack #comment\r\n";
+    File.close();
+  }
+  sycl::detail::readConfig(true);
+  EXPECT_EQ(std::string("after_addCopyBack"),
+            sycl::detail::SYCLConfig<
+                sycl::detail::SYCL_PRINT_EXECUTION_GRAPH>::get());
+}
```

Comment:
Please add an empty line at the end of file

Reply:
Fixed